lib/kinesis_client.rb

```diff
@@ -84,19 +78,31 @@ def push_batch(batch)
 
     return_message = {
       failures: resp.failed_record_count,
-      error_messages: resp.records.map {|record| record.error_message }.compact
+      failures_data: filter_failures(resp),
+      # error_messages: resp.records.map { |record| record.error_message }.compact
     }
 
     $logger.info("Message sent to #{config[:stream_name]} #{return_message}") if $logger
 
-    return {
+    {
       "code": "200",
       "message": return_message.to_json
     }
   end
 
   def push_records
-    @batch.each_slice(@batch_size) {|slice| push_batch slice}
-    @batch = []
+    @records.each_slice(@batch_size) { |slice| push_batch(slice) } if @records.length > 0 
+    @records = []
+  end
+
+  def filter_failures(resp)
+    failed_records = []
+    resp.records.each_with_index do |record, i|
+      if record.key?(:error_message)
```

Comment:
The record isn't actually a Hash, right? Are you sure about this way of checking for the presence of an error message?